test/functional/Jsr292/src/com/ibm/j9/jsr292/FilterArgumentsWithCombinerTest.java

```diff
@@ -0,0 +1,435 @@
+    private final static int constInt1 = (int) 1213988863;
+    private final static int constInt2 = (int) -1993736192;
+    private final static long constLong1 = (long) 4.21368040135852e+018;
+    private final static long constLong2 = (long) 1.26663739519795e+018;
+    private final static float constFloat1 = (float) 6.12483306976318e+037F;
+    private final static float constFloat2 = (float) 1.53640056404114e+038F;
+    private final static double constDouble1 = (double) 7.33110759312901e+307;
+    private final static double constDouble2 = (double) 1.10819706189633e+307;
+    private final static boolean constBoolMax = true;
+    private final static boolean constBoolMin = false;
+
+    private final static String MHClassName = "java.lang.invoke.MethodHandles";
+    private static Class<?> mhClazz;
+    private static Method fawcMethod;
+
+    static {
+        try {
+            mhClazz = Class.forName(MHClassName);
+            fawcMethod = mhClazz.getDeclaredMethod("filterArgumentsWithCombiner", MethodHandle.class, int.class, MethodHandle.class, int[].class);
+            fawcMethod.setAccessible(true);
+        } catch(Throwable e) {
+            AssertJUnit.fail("FilterArgumentsWithCombinerTest setup failure" + e.getMessage());
+        }
+    }
+
+    /* test filterArgumentsWithCombiner for data type byte */
+    @Test(groups = { "level.extended" })
+	public void test_filterArgumentsWithCombiner_Byte() throws WrongMethodTypeException, Throwable {
+        MethodHandle result;
+
+        /* handle & filter */
+        MethodHandle target = MethodHandles.publicLookup().findStatic(FilterArgumentsWithCombinerTest.class, 
+            "targetByte", MethodType.methodType(byte.class, byte.class, byte.class, byte.class));
+        MethodHandle combiner = MethodHandles.publicLookup().findStatic(FilterArgumentsWithCombinerTest.class, 
+            "combinerByte", MethodType.methodType(byte.class, byte.class, byte.class));
+
+        /* verify maximum boundary, filter position is the middle handle parameter */
+        result = (MethodHandle)fawcMethod.invoke(mhClazz, target, 1, combiner, new int[]{2, 0});
+        AssertJUnit.assertEquals(Byte.MAX_VALUE, (byte)result.invokeExact(Byte.MAX_VALUE, constByte2, Byte.MAX_VALUE));
+
+        /* verify minimum boundary, filter position is first handle parameter */
+        result = (MethodHandle)fawcMethod.invoke(mhClazz, target, 0, combiner, new int[]{0, 2}); 
+        AssertJUnit.assertEquals(Byte.MIN_VALUE, (byte)result.invokeExact(constByte1, Byte.MIN_VALUE, constByte2));
+
+        /* verify medium boundary, filter position is last handle parameter */
+        result = (MethodHandle)fawcMethod.invoke(mhClazz, target, 2, combiner, new int[]{1, 2});
+        AssertJUnit.assertEquals(constByte1, (byte)result.invokeExact(Byte.MAX_VALUE, constByte1, constByte2));
+    }
+
+    /* combiner returns highest byte value */
+    public static byte combinerByte(byte v1, byte v2) {
```

Comment:
All test methods like this should be moved to a separate file as this file only contains test cases.